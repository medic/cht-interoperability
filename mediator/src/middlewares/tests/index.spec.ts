import Joi from "joi";
import { validateBodyAgainst } from "..";

describe("validateBodyAgainst", () => {
  const schema = Joi.object({ name: Joi.string().required() }).required();
  let mockRes: any = {};

  beforeEach(() => {
    mockRes.send = jest.fn();
    mockRes.status = jest.fn(() => mockRes);
  })

  it("validates the request and passes control over to the next function", async () => {
    const mockReq = {body: { name: "John Doe"}};
    const mockNext = jest.fn();

    const handler = validateBodyAgainst(schema)
    await handler(mockReq as any, mockRes as any, mockNext);

    expect(mockNext).toHaveBeenCalled();
    expect(mockRes.send).not.toHaveBeenCalled();
    expect(mockRes.status).not.toHaveBeenCalled();
  })

  it("fails to validate when given a non schema as input", async () => {
    const mockReq = {body: { name: "John Doe"}};
    const mockNext = jest.fn();

    const handler = validateBodyAgainst(undefined)
    expect(() => handler(mockReq as any, mockRes as any, mockNext)).toThrowErrorMatchingSnapshot();
  })

  it("sends and error and a 400 status code when the request body is bad", async () => {
    const mockReq = {body: { }};
    const mockNext = jest.fn();

    const handler = validateBodyAgainst(schema)
    await handler(mockReq as any, mockRes as any, mockNext);

    expect(mockNext).not.toHaveBeenCalled();
    expect(mockRes.status).toHaveBeenCalled();
    expect(mockRes.send).toHaveBeenCalled();
    expect(mockRes.send).toMatchSnapshot();
  })
})
