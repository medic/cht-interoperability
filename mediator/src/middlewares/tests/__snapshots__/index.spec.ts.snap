// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validateBodyAgainst fails to validate when given a non schema as input 1`] = `"Cannot read properties of undefined (reading 'validateAsync')"`;

exports[`validateBodyAgainst sends and error and a 400 status code when the request body is bad 1`] = `
[MockFunction] {
  "calls": [
    [
      {
        "message": ""name" is required",
      },
    ],
  ],
  "results": [
    {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
